File: src/main/java/fr/lewon/bot/manager/service/BotService.java
package fr.lewon.bot.manager.service;

import java.util.Map;

import org.springframework.core.io.ClassPathResource;

import fr.lewon.bot.manager.entities.BotInfosDTO;
import fr.lewon.bot.manager.entities.BotLogsDTO;
import fr.lewon.bot.manager.entities.BotMethodsDTO;
import fr.lewon.bot.manager.entities.BotPropertiesDescriptorsDTO;
import fr.lewon.bot.manager.entities.GameInfosListDTO;
import fr.lewon.bot.manager.util.errors.BotManagerException;

public interface BotService {

	void createBot(String login, String password, String gameId, Map<String, Object> params) throws BotManagerException;

	void stopBot(String gameId, Long id) throws BotManagerException;

	void pauseBot(String gameId, Long id) throws BotManagerException;
	
	void startBot(String gameId, Long id) throws BotManagerException;

	GameInfosListDTO getAllBotInfos() throws BotManagerException;

	void trimStoppedBots() throws BotManagerException;

	void trimStoppedBot(String gameId, Long id) throws BotManagerException;

	BotLogsDTO getBotLogs(String gameId, Long id) throws BotManagerException;

	BotMethodsDTO getMethods(String gameId, Long botId) throws BotManagerException;

	Object callMethod(String gameId, String idMethod, Long botId, Map<String, Object> params) throws BotManagerException;

	ClassPathResource getIcon(String gameId) throws BotManagerException;

	BotPropertiesDescriptorsDTO getProperties(String gameId) throws BotManagerException;

	BotInfosDTO getBotInfos(String gameId, Long id) throws BotManagerException;

}

File: src/main/java/fr/lewon/bot/manager/service/BotService.kt
package fr.lewon.bot.manager.service

import fr.lewon.bot.manager.entities.*
import fr.lewon.bot.manager.util.errors.BotManagerException
import fr.lewon.bot.runner.bot.operation.OperationResult
import org.springframework.core.io.ClassPathResource

interface BotService {
    @Throws(BotManagerException::class)
    fun createBot(login: String, password: String, gameId: Long, params: Map<String, String?>)

    @Throws(BotManagerException::class)
    fun stopBot(id: Long)

    @Throws(BotManagerException::class)
    fun startBot(id: Long)

    @Throws(BotManagerException::class)
    fun getAllBotInfo(): GameInfoListDTO

    @Throws(BotManagerException::class)
    fun trimStoppedBots()

    @Throws(BotManagerException::class)
    fun trimStoppedBot(id: Long)

    @Throws(BotManagerException::class)
    fun getBotLogs(id: Long): BotLogsDTO

    @Throws(BotManagerException::class)
    fun getBotOperations(botId: Long): BotOperationsDTO

    @Throws(BotManagerException::class)
    fun callBotOperation(operationId: Long, botId: Long, params: Map<String, String?>): OperationResult

    @Throws(BotManagerException::class)
    fun getIcon(gameId: Long): ClassPathResource

    @Throws(BotManagerException::class)
    fun getGameProperties(gameId: Long): BotPropertiesDescriptorsDTO

    @Throws(BotManagerException::class)
    fun getBotProperties(id: Long): BotPropertiesDescriptorsDTO

    @Throws(BotManagerException::class)
    fun getBotInfo(id: Long): BotInfoDTO
}